### PR TITLE
Eric db reuse all in one

### PR DESCRIPTION
**Pull request by ewollesen**
All my db PRs in one, as requested.

### CODE REVIEW

File: data/store/mongo/suite_store_test.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package mongo_test
```
Comment (darinkrauss) [inline data/store/mongo/suite_store_test.go:1]: For consistency sake, please rename to `mongo_suite_test.go`. It is `<package>_suite_test.go`.

File: data/summary/store/store_test.go [hunk 2]
```diff
@@ -26,53 +26,59 @@ var _ = Describe("Summary Stats Mongo", Label("mongodb", "slow", "integration"),
...
-
-	AfterEach(func() {
-		if store != nil {
-			_ = store.Terminate(context.Background())
-		}
 	})
 
 	Context("Create Stores", func() {
+		var config *storeStructuredMongo.Config
+		var createStore *dataStoreMongo.Store
```
Comment (darinkrauss) [inline data/summary/store/store_test.go:40]: Why not just use `store`, declared above?

File: prescription/store/mongo/store_test.go [hunk 3]
```diff
@@ -88,3 +67,16 @@ var _ = Describe("Store", Label("mongodb", "slow", "integration"), func() {
 		})
 	})
 })
+
+var suiteStore *prescriptionStoreMongo.PrescriptionStore
```
Comment (darinkrauss) [inline prescription/store/mongo/store_test.go:71]: I'd strongly recommend adding this "common" code in a common location like the `*_suite_test.go` file for each package. That way, if anyone wants to find the suite store then just need to look in one place.

**Review by darinkrauss — COMMENTED** [review]
A few comments and questions, plus some suggestions.

**Comment by darinkrauss** [comment]
Oh, plus the comments/suggestions from https://github.com/tidepool-org/platform/pull/772.

File: store/structured/mongo/test/test.go [hunk 4]
```diff
@@ -23,14 +25,39 @@ var _ = ginkgo.BeforeSuite(func() {
...
+		gomega.Expect(err).To(gomega.Succeed())
+		gomega.Expect(store).ToNot(gomega.BeNil())
+		suiteStore = store
+	})
+	return suiteStore
+}
+
 var _ = ginkgo.AfterSuite(func() {
+	ctx := context.Background()
+	dropTestDatabase(ctx)
```
[the remark this answers is not in the document]
Comment (darinkrauss) [reply store/structured/mongo/test/test.go:46]: No, I suppose not. If you maintained the same order (drop database *then* terminate store) the test should function fine. I guess it just makes sense to me to terminate the store first before dropping the database. That is, terminating the store is part of the code you are testing while dropping the database is part of the test "framework".

File: blob/store/structured/mongo/mongo_test.go [hunk 5]
```diff
@@ -2,127 +2,141 @@ package mongo_test
...
 
 var _ = Describe("Mongo", func() {
-	var config *storeStructuredMongo.Config
-	var store *blobStoreStructuredMongo.Store
-
-	BeforeEach(func() {
-		config = storeStructuredMongoTest.NewConfig()
-	})
+	Context("NewStore", func() {
+		var createStore *blobStoreStructuredMongo.Store
```
[the remark this answers is not in the document]
Comment (darinkrauss) [reply blob/store/structured/mongo/mongo_test.go:21]: Oh, I see. I would have just named it `store` given the context is different, but that's fine.

**Review by darinkrauss — APPROVED** [review]
LGTM!

**Comment by ewollesen** [comment]
Can you spot which jobs were testing this branch?

```
job: 7456 result: 12m2.176390313s
job: 7455 result: 12m13.751922585s
job: 7454 result: 11m49.86379335s
job: 7453 result: 11m47.829861278s
job: 7452 result: 11m48.06876532s
job: 7451 result: 11m59.215935727s
job: 7450 result: 11m43.947634279s
job: 7448 result: 6m32.469842118s
job: 7447 result: 6m37.686044871s
job: 7446 result: 6m34.628773662s
job: 7445 result: 6m25.416739443s
job: 7444 result: 6m38.024252853s
job: 7443 result: 6m30.37500431s
```

**Pull request merged**